base/log.go

```diff
@@ -31,24 +43,32 @@ func DatasetLog(ctx context.Context, r repo.Repo, ref reporef.DatasetRef, limit,
 					if local {
 						if ds, err := dsfs.LoadDataset(ctx, r.Store(), v.Path); err == nil {
 							if ds.Commit != nil {
-								versions[i].CommitMessage = ds.Commit.Message
+								items[i].CommitMessage = ds.Commit.Message
+								items[i].CommitTitle = ds.Commit.Title
+								items[i].CommitTime = ds.Commit.Timestamp
 							}
 						}
 					}
 					versions[i].Foreign = !local
+					items[i].VersionInfo = versions[i]
 				}
 			}
-			return versions, nil
+			return items, nil
 		}
 	}
 
 	rlog, err := DatasetLogFromHistory(ctx, r, ref, offset, limit, loadDatasets)
 	if err != nil {
 		return nil, err
 	}
-	items := make([]dsref.VersionInfo, len(rlog))
+	items := make([]DatasetLogItem, len(rlog))
 	for i, vref := range rlog {
-		items[i] = reporef.ConvertToVersionInfo(&vref)
+		items[i].VersionInfo = reporef.ConvertToVersionInfo(&vref)
```

Comment:
I wanted to extract this peice, but since DataLogItem is too high up in base, couldn't drop the login into the repo/ref/converter.

Reply:
Just for future reference, this is me being wrong with some assumptions. Refer to https://github.com/qri-io/qri/issues/1144 for a better overview of the components and it's interactions.